sphinx/parsers.py

```diff
@@ -58,12 +58,21 @@ def set_application(self, app):
 class RSTParser(docutils.parsers.rst.Parser):
     """A reST parser customized for Sphinx."""
 
+    smartquotes_allowed = True
+
+    def set_smartquotes_action(self, action):
```

Comment:
Let's make `RSTParser` class a subclass of `sphinx.parsers.Parser`. After that we can refer config object by `self.config`.
I think storing status to class variable is not good way. Because it behaves like global variables.

Reply:
agreed, done (will comment after push)